### PR TITLE
feat(cli): `--exclude` flag for ignoring files

### PR DESCRIPTION
**Pull request by tomaskudlicka**
I added flag for excluding paths or files.
https://github.com/Nutlope/aicommits/issues/128

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -24,6 +24,11 @@ cli(
 				description: 'Number of messages to generate. (Warning: generating multiple costs more) (default: 1)',
 				alias: 'g',
 			},
+			filename: {
```
Comment (privatenumber) [inline src/cli.ts:27]:
```suggestion
			exclude: {
```

File: src/cli.ts [hunk 2]
```diff
@@ -24,6 +24,11 @@ cli(
...
+				type: String,
+				description: 'Excluding a path.',
```
Comment (privatenumber) [inline src/cli.ts:29]:
```suggestion
				description: 'Paths to exclude.'
```

File: src/cli.ts [hunk 3]
```diff
@@ -24,6 +24,11 @@ cli(
...
+				alias: 'f',
```
Comment (privatenumber) [inline src/cli.ts:30]:
```suggestion
				alias: 'e',
```

Or maybe `x`. Is there a common precedent?

File: src/commands/aicommits.ts [hunk 4]
```diff
@@ -14,8 +14,9 @@ import { getConfig } from '../utils/config.js';
 import { generateCommitMessage } from '../utils/openai.js';
 import { KnownError, handleCliError } from '../utils/error.js';
 
-export default async (
+export default async (:
```
Comment (privatenumber) [inline src/commands/aicommits.ts:17]: ?

File: src/commands/aicommits.ts [hunk 5]
```diff
@@ -16,15 +16,18 @@ import { KnownError, handleCliError } from '../utils/error.js';
 
 export default async (
 	generate: number | undefined,
+	excludeFile: string[] | undefined,
 	rawArgv: string[],
 ) => (async () => {
 	intro(bgCyan(black(' aicommits ')));
-
+	console.log(excludeFile)
```
Comment (privatenumber) [inline src/commands/aicommits.ts:23]: Remove

File: src/commands/aicommits.ts [hunk 6]
```diff
@@ -16,15 +16,18 @@ import { KnownError, handleCliError } from '../utils/error.js';
...
 	await assertGitRepo();
 
 	const detectingFiles = spinner();
 	detectingFiles.start('Detecting staged files');
-	const staged = await getStagedDiff();
+	const staged = await getStagedDiff(
+		excludeFile,
+	);
```
Comment (privatenumber) [inline src/commands/aicommits.ts:30]: Single line

File: src/commands/aicommits.ts (hunk 5 above)
Comment (tomaskudlicka) [reply src/commands/aicommits.ts:23]: sorry i forgot

File: src/cli.ts [hunk 7]
```diff
@@ -36,13 +41,15 @@ cli(
 		},
 
 		ignoreArgv: type => type === 'unknown-flag' || type === 'argument',
+		//
```
Comment (privatenumber) [inline src/cli.ts:44]: Remove.

**Pull request merged**